### ISSUE TITLE
elb_target_group fails with "KeyError: 'Port'" on target group creation

### ISSUE THREAD

**Issue by jrko** [ansible/ansible#38564]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

elb_target_group

##### ANSIBLE VERSION

```
ansible 2.5.0
  config file = /home/vagrant/repo/v2-ansible/ansible.cfg
  configured module search path = [u'/home/vagrant/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.13 (default, Nov 24 2017, 17:33:09) [GCC 6.3.0 20170516]

```

##### CONFIGURATION

```
ANSIBLE_SSH_ARGS(/home/vagrant/repo/v2-ansible/ansible.cfg) = -C -o ControlMaster=auto -o ControlPersist=60s -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no
DEFAULT_HOST_LIST(/home/vagrant/repo/v2-ansible/ansible.cfg) = [u'/home/vagrant/repo/v2-ansible/environment/test']
DEFAULT_REMOTE_USER(/home/vagrant/repo/v2-ansible/ansible.cfg) = {{ ansible_username | default('ansible') }}
HOST_KEY_CHECKING(/home/vagrant/repo/v2-ansible/ansible.cfg) = False
RETRY_FILES_ENABLED(/home/vagrant/repo/v2-ansible/ansible.cfg) = False

```
##### OS / ENVIRONMENT

Debian 9.3 
`Linux stretch 4.9.0-4-amd64 #1 SMP Debian 4.9.65-3 (2017-12-03) x86_64 GNU/Linux`

##### SUMMARY

elb_target_group module fails to create/update target group.

##### STEPS TO REPRODUCE

Create ELB target group and specify the `port` parameter with `state: present`.

```yaml
- name: Create/update Foo ELB target group
  elb_target_group:
    name: "{{ foo_name }}"
    protocol: http
    port: "{{ foo_port }}"
    targets: "{{ foo_targets }}"
    tags:
      Name: "{{ foo_name }}"
    vpc_id: "{{ vpc.vpcs.0.id }}"
    region: "{{ ec2_region }}"
    state: present
```
my `{{ foo_targets }}` looks like:
```
ok: [localhost] => {
    "ansible_facts": {
        "foo_targets": [
            {
                "Id": "i-0ee19f7966cf30056"
            }, 
            {
                "Id": "i-094756058d65990da"
            }
        ]
    }, 
    "changed": false
}
```

##### EXPECTED RESULTS

The ELB target group gets created. This was working in 2.4.3.

##### ACTUAL RESULTS

MODULE FAILURE

```
The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_GCRxtK/ansible_module_elb_target_group.py", line 717, in <module>
    main()
  File "/tmp/ansible_GCRxtK/ansible_module_elb_target_group.py", line 712, in main
    create_or_update_target_group(connection, module)
  File "/tmp/ansible_GCRxtK/ansible_module_elb_target_group.py", line 505, in create_or_update_target_group
    instances_to_add.append({'Id': target['Id'], 'Port': int(target['Port'])})
KeyError: 'Port'

fatal: [localhost]: FAILED! => {
    "changed": false, 
    "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_GCRxtK/ansible_module_elb_target_group.py\", line 717, in <module>\n    main()\n  File \"/tmp/ansible_GCRxtK/ansible_module_elb_target_group.py\", line 712, in main\n    create_or_update_target_group(connection, module)\n  File \"/tmp/ansible_GCRxtK/ansible_module_elb_target_group.py\", line 505, in create_or_update_target_group\n    instances_to_add.append({'Id': target['Id'], 'Port': int(target['Port'])})\nKeyError: 'Port'\n", 
    "module_stdout": "", 
    "msg": "MODULE FAILURE", 
    "rc": 1
}

```

**Comment by ansibot** [comment]
Files identified in the description:
* [lib/ansible/modules/cloud/amazon/elb_target_group.py](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/elb_target_group.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @willthames @wimnat
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by willthames** [comment]
Can you confirm by running this code against 2.4.3 now? - nothing in the code appears to have changed in this area.

It looks like your `foo_targets` should both contain a `Port` key, and that's why you're getting this error.

**Comment by jrko** [comment]
Sorry, it looks like this was working in `2.4.1`, not `2.4.3` as I stated. It also fails in `2.4.2`, so it appears the changes are there.

I can try and append the `Port` key to my targets, but what about the module's `port` parameter then?

**Comment by willthames** [comment]
Certainly that's what the docs suggest should happen, you're right. 

This is definitely a bug - providing `Port` will offer a workaround until it's fixed, but we should restore 2.4.1 behaviour. Thanks for clarifying the version.

**Comment by willthames** [comment]
@jrko can you test if #38578 fixes this issue?

**Comment by jrko** [comment]
@willthames I'm seeing some strange behavior. Yesterday I couldn't reproduce the issue. I deleted the target group and could create a new one in 2.5.0 and 2.4.3 without errors.

Today I just now rebuilt VPC from scratch and launched new instances. When I tried to create the target group, both 2.5.0 and 2.4.3 fail with the Port KeyError.

I changed line 505 as per your PR, and the target group gets created.

However, if I revert the change to line 505, delete the target group, and try again, the target group gets created without error! So it seems the issue appears only when the VPC has just been built and no target group has yet existed within it.

So, your PR seems to fix the issue, with this additional behavior observed.

EDIT: I just built another VPC and wanted to confirm the additional behavior I described above is reproducible 3 times in a row. The error only appears on initial provisioning.
